### ISSUE TITLE
[Feature request] Microdata/JSON-LD Support

### ISSUE THREAD

**Issue by andreacanton** [lorisleiva/vuepress-plugin-seo#6]
Hi!

I think it should be supported the Microdata/JSON-LD.

More info about this structured data here:

https://developers.google.com/search/docs/guides/intro-structured-data?hl=en

**Comment by lorisleiva** [comment]
Hi there 👋

Good idea.

The challenge for me here was to find a flexible API to provide that JSON schema.

PR welcome. 🙂

**Comment by adamdehaven** [comment]
@lorisleiva I can help with this, but a question: how would you inject the structured data into the rendered page, without JavaScript so that it's available on page load?

**Comment by lorisleiva** [comment]
Vuepress compiles for both server-side rendering and client-side rendering. So, just like we do with the current metadata, we just need to make sure it's applied when compiling the initial HTML pages.

**Comment by adamdehaven** [comment]
Right, but I'm not sure the way it's currently implemented works. The data is only correct for the first page load and doesn't seem to update when moving to other pages.

**Comment by adamdehaven** [comment]
Right, but you can't inject a script tag as part of the page meta or frontmatter, so... How would you do that?

**Comment by lorisleiva** [comment]
I'm hoping the `innerHTML` argument could do the trick but otherwise, we'd have to manually inject it somehow by looking into how that `head` configuration work on the Vuepress core.

<img width="796" alt="Screenshot 2020-08-15 at 18 20 36" src="https://user-images.githubusercontent.com/3642397/90317828-07267900-df24-11ea-8639-38167117f8b5.png">

**Comment by adamdehaven** [comment]
@lorisleiva when you pass values there, they don't update on route change though ...

**Comment by amerkay** [comment]
Check out this [article about adding Canonical, Meta Tags, and Structured Data](https://www.adamdehaven.com/blog/how-to-add-metadata-canonical-urls-and-structured-data-to-your-vuepress-site/#create-structured-data-component).

We should implement for VuePress v2. I already submitted a pull request with upgraded code to v2.
